// jest-stare/lib/utils/IO.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const fs = require("fs");
const pkgUp = require("pkg-up");
class IO {
    static unlinkSync(file) {
        if (IO.existsSync(file)) {
            fs.unlinkSync(file);
        }
    }
    static writeFile(path, data) {
        return new Promise((resolve, reject) => {
            fs.writeFile(path, data, (error) => {
                if (error) {
                    reject(error);
                }
                resolve();
            });
        });
    }
    static writeFileSync(path, data) {
        fs.writeFileSync(path, data);
    }
    static mkDirSync(dir) {
        if (!fs.existsSync(dir)) {
            fs.mkdirSync(dir);
        }
    }
    static mkdirsSync(dir) {
        let built = "";
        const dirs = dir.split("/");
        dirs.forEach((d) => {
            built += (d + "/");
            IO.mkDirSync(built);
        });
    }
    static readFileSync(path) {
        return fs.readFileSync(path).toString();
    }
    static existsSync(path) {
        return fs.existsSync(path);
    }
    static readPackageJson() {
        const packageJson = pkgUp.sync();
        if (packageJson !== null) {
            return JSON.parse(IO.readFileSync(packageJson));
        }
        else {
            return {};
        }
    }
    static copyFileSync(src, dest) {
        fs.copyFileSync(src, dest);
    }
}
exports.IO = IO;
